Yoast/wordpress-seo
# Add tests for schema webpage generator

## Context


* We want to have all the code covered in unit tests.

## Summary


This PR can be summarized in the following changelog entry:

* N/A Adds unit tests for the Schema webpage generator.

## Relevant technical choices:

* I used a provider to test all the scenarios in which the exact same functions are called. 
* Not all the variables could be part of this provider. For example, I couldn't pass `description` to the provider, because when `description` is not empty, the function `smart_strip_tags` is called, and otherwise it isn't. This means that you cannot cover both scenarios with the same test.
* I tested all the possible paths by adding tests for each scenario in each conditional.

## Test instructions

This PR can be tested by following these steps:

* Run the unit tests and see that they pass.

## UI changes

## Documentation

## Quality assurance


Fixes #14720
